src/NBS.Appointments.Service.Core/Services/QflowService.cs

```diff
@@ -57,7 +67,16 @@ public class QflowService : IQflowService
             var response = await Execute(query, endpointUrl, HttpMethod.Get, null);
             var responseBody = await response.Content.ReadAsStringAsync();
 
-            return JsonSerializer.Deserialize<SiteAvailabilityResponse[]>(responseBody);
+            try
+            {
+                return JsonSerializer.Deserialize<SiteAvailabilityResponse[]>(responseBody);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Exception thrown when attempting to deserialize qflow site availability response. Message: {@Message}, Response body: {@ResponseBody}, StatusCode: {@StatusCode}",
+                    ex.Message, responseBody, response.StatusCode);
+                throw;
```

Comment:
Do we want to throw here or change the response model to return a non-successful status code to NBS? Similar to some other methods further below.